records.py

```
@@ -306,7 +322,33 @@ def get_connection(self, close_with_result=False):
+    @contextmanager
+    def transaction(self) -> Generator['Connection', None, None]:
+        """Create a database transaction context manager that automatically
+        commits on success or rolls back on error.
+        
+        Usage:
+            with db.transaction() as conn:
+                conn.query("INSERT INTO table VALUES (?)", value=123)
+                # Transaction is automatically committed here
+        """
+        if not self.open:
+            raise exc.ResourceClosedError("Database closed.")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)